### PR TITLE
perf(bootstrap): Lazy register template creator through event

### PR DESCRIPTION
**Pull request by juliushaertl**
Requires https://github.com/nextcloud/server/pull/44952

This will avoid triggering dependency injection queries during app bootstrap for every request.

### CODE REVIEW

File: lib/Listeners/RegisterTemplateCreatorListener.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2020 Morris Jobke <hey@morrisjobke.de>
+ *
+ * @author Morris Jobke <hey@morrisjobke.de>
```
Comment (max-nextcloud) [inline lib/Listeners/RegisterTemplateCreatorListener.php:7]: Probably outdated, right?

We might also start using the SPDX headers right away when adding new files.

**Review by max-nextcloud — COMMENTED** [review]
Approach seems good - for what i understand. Linter and Static check still failing.

File: lib/Listeners/RegisterTemplateCreatorListener.php (hunk 1 above)
Comment (juliushaertl) [reply lib/Listeners/RegisterTemplateCreatorListener.php]: Adjusted the copyright, but would rather keep moving to SPDX separate and do it with the migration script from Joas then for all files

**Comment by hweihwang** [comment]
Looks good, smart take on deferring the template creator setup until it is actually needed

**Review by hweihwang — APPROVED** [review]

**Pull request merged**